pkg/webui/console/views/device-add/wizard/network-settings-form/index.js

```diff
@@ -68,6 +71,26 @@ const NetworkSettingsForm = props => {
     setResetsFCnt(checked)
   }, [])
 
+  const formRef = React.useRef(null)
+  const fCntWidthEncode = React.useCallback(value => {
+    return value === FRAME_WIDTH_COUNT.SUPPORTS_32_BIT
+  }, [])
+  const fCntWidthDecode = React.useCallback(value => {
+    return value ? FRAME_WIDTH_COUNT.SUPPORTS_32_BIT : FRAME_WIDTH_COUNT.SUPPORTS_16_BIT
+  }, [])
+  const handleFrameCounterWidthChange = React.useCallback(evt => {
+    const { values, setValues } = formRef.current
+    setValues(
+      validationSchema.cast({
+        ...values,
+        mac_settings: {
+          ...values.mac_settings,
+          supports_32_bit_f_cnt: evt,
+        },
+      }),
+    )
+  }, [])
```

Comment:
```suggestion
```
This shouldn't be necessary. After checking the `<Field />` component, I realized there's a bug in there that causes the form to ignore the encoder. Please apply this diff
```diff
diff --git a/pkg/webui/components/form/field/index.js b/pkg/webui/components/form/field/index.js
index 6b7206a27..ae21156a0 100644
--- a/pkg/webui/components/form/field/index.js
+++ b/pkg/webui/components/form/field/index.js
@@ -122,7 +122,8 @@ class FormField extends React.Component {
     const { setFieldValue, setFieldTouched } = this.context
 
     // Check if the value is react's synthetic event.
-    const newValue = this.extractValue(value)
+    const isSyntheticEvent = typeof value === 'object' && 'target' in value
+    const newValue = encode(this.extractValue(value))
 
     // TODO: Remove `await` and event persist when https://github.com/jaredpalmer/formik/issues/2457
     // is resolved.
@@ -136,7 +137,7 @@ class FormField extends React.Component {
       setFieldTouched(name)
     }
 
-    onChange(encode(value))
+    onChange(isSyntheticEvent ? value : encode(value))
   }
 
   @bind

```